app/Http/Controllers/BookingAPIController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Booking;
+use App\Models\Room;
+use Carbon\CarbonPeriod;
+use http\Env\Response;
+use Illuminate\Http\Request;
+
+class BookingAPIController extends Controller
+{
+    public function create(Request $request)
+    {
+
+        $request->validate([
+            'room_id' => 'required|integer|exists:rooms,id',
+            'customer' => 'required|string|max:255',
+            'guests' => 'required|integer|min:1',
+            'start' => 'required|date|date_format:Y-m-d|after_or_equal:today',
+            'end' => 'required|date|date_format:Y-m-d|after:start',
+        ]);
+
+        $room = Room::find($request->room_id);
+        $startDate = $request->start;
+        $endDate = $request->end;
+
+
+        $bookingConflict = Booking::where('room_id', $room->id)
+            ->where(function ($query) use ($startDate, $endDate) {
+                $query->whereBetween('start', [$startDate, $endDate])
+                    ->orWhereBetween('end', [$startDate, $endDate])
+                    ->orWhere(function ($query) use ($startDate, $endDate) {
+                        $query->where('start', '<=', $startDate)
+                            ->where('end', '>=', $endDate);
+                    });
+            })
+            ->exists();
+
+        if ($room->min_capacity > $request->guests OR  $room->max_capacity < $request->guests) {
+            return response()->json([
+                'message' => "The {$room->name} room can only accommodate between {$room->min_capacity} and {$room->max_capacity} guests",
+            ],400);
+        }
+        elseif ($bookingConflict)
+        {
+                return response()->json([
+                    'message' => "The {$room->name} room is unavailable for the chosen dates."
+                ], 400);
+        }
+        else
```

Comment:
You can drop this else now.

if (guests are wrong check) {
    return error response
}

if (booking conflict) {
    return error response
}

make the booking 

return success response

Reply:
This is a practice often called early returning - it can significantly reduce the complexity of the code by having small individual if statements vs a big if/elseif/else (that will inevitably get more complex in the future) that can be quite hard to follow